snapshots/manager.go

```diff
@@ -178,8 +241,62 @@ func (m *Manager) Restore(snapshot types.Snapshot) error {
 	chChunks := make(chan io.ReadCloser, chunkBufferSize)
 	chReady := make(chan struct{}, 1)
 	chDone := make(chan restoreDone, 1)
+
+	doRestore := func() error {
```

Comment:
I'm rather against the big "inner" clousure functions. They are impossible to test.
Let's create a struct with the readers and other channels, and a method instead of this clousure.

Reply:
I think we can also refactor it into a method, and pass the channels as parameter.

Reply:
done